frontend/src/metabase-query/lib/query.js

```diff
@@ -0,0 +1,83 @@
+// This is meant to be an encapsulation layer around MBQL query dictionaries
+// This can be fairly complicated internally, but the consumer of this api
+// shouldn't know anything about our metadata formats
+
+
+class Query {
+    constructor(metadata: Metadata, datasetQuery: DatasetQuery) {}
+
+    setDatabase(database: Database) {}
+    setTable(table: Table) {}
+
+    aggregations(): Aggregation[] {}
+    aggregationOptions(): AggregationOption[] {}
+    canAddAggregation(): bool {}
+
+    breakouts(): Breakout[] {}
+    breakoutOptions(unused: boolean = false): BreakoutOption[] {}
+    canAddBreakout(): bool {}
+
+    filters(): Filter[] {}
+    filterOptions(): FilterOption[] {}
+    canAddFilter(): bool {}
+
+    sorts(): Sort[] {}
+    sortOptions(): SortOption[] {}
+    canAddSort(): bool {}
+
+    addLimit(int: limit){}
+
+    // SQL 
+    getSQL(){}
+    convertToSQL(){}
```

Comment:
is this sync or async? it would need the results of running the query which includes the native query (or some other API call)